src/components/text-editor/__internal__/text-editor.component.js

```diff
@@ -0,0 +1,273 @@
+import React, {
+  useCallback, useEffect, useRef, useState
+} from 'react';
+import PropTypes from 'prop-types';
+import {
+  ContentState,
+  EditorState,
+  Editor,
+  RichUtils,
+  getDefaultKeyBinding
+} from 'draft-js';
+import {
+  computeBlockType,
+  getContent,
+  getContentInfo,
+  getDecoratedValue,
+  getSelectedLength,
+  moveSelectionToEnd,
+  resetBlockType,
+  isASCIIChar,
+  replaceText,
+  hasInlineStyle,
+  hasBlockStyle,
+  blockStyleFn
+} from './utils';
+import StyledEditorContainer from './text-editor.style';
+import Counter from './editor-counter';
+import Toolbar from './toolbar';
+import Label from '../../../__experimental__/components/label';
+import Events from '../../../utils/helpers/events/events';
+
+const NUMBERS = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+const INLINE_STYLES = ['BOLD', 'ITALIC'];
+const BLOCK_TYPES = ['unordered-list-item', 'ordered-list-item'];
+
+const TextEditor = React.forwardRef(({
+  characterLimit,
+  labelText,
+  labelId,
```

Comment:
Don't you think that this `labelId` prop is not necessary anymore? I think there's no added value of having this id defined by the user.
It could be defined internally as `const labelId = useRef(guid());`